src/xtgeo/xyz/_xyz_roxapi.py

```diff
@@ -343,7 +379,59 @@ def _get_roxitem(self, proj, name, category, stype, mode="set"):  # pragma: no c
                 roxxyz = proj.clipboard.create_points(name, folders)
 
     else:
-        roxxyz = None
         raise ValueError("Unsupported stype: {}".format(stype))
 
     return roxxyz
+
+
+def _get_roxxyz(
+    rox, name, category, stype, mode="set", is_polygons=False
+):  # pragma: no cover
+    # pylint: disable=too-many-branches
+    """Get the correct rox_xyz which is some pointer to a RoxarAPI structure."""
+    if stype == "horizons":
+        rox_xyz = rox.project.horizons[name][category]
+    elif stype == "zones":
+        rox_xyz = rox.project.zones[name][category]
+    elif stype == "faults":
+        rox_xyz = rox.project.faults[name][category]
+    elif stype == "clipboard":
+        folders = None
+        rox_xyz = rox.project.clipboard
+        if category:
+            if isinstance(category, list):
+                folders = category
+                folders.append(category)
+            else:
+                raise RuntimeError(
+                    f"Cannot parse category: {category}, see documentation!"
+                )
```

Comment:
I think you need to add before the else block:
```
            elif isinstance(category, str) and "|" in category:
                folders = category.split("|")
            elif isinstance(category, str) and "/" in category:
                folders = category.split("/")
            elif isinstance(category, str):
                folders = []
                folders.append(category)
```

Reply:
Not sure I follow what is going on there, do you have an example?